Tests/DependencyInjection/Factory/Resolver/AwsS3ResolverFactoryTest.php

```diff
@@ -107,8 +107,12 @@ public function testCreateS3ClientDefinitionOnCreate()
 
         $clientDefinition = $container->getDefinition('liip_imagine.cache.resolver.theresolvername.client');
         $this->assertEquals('Aws\S3\S3Client', $clientDefinition->getClass());
-        $this->assertEquals('Aws\S3\S3Client', $clientDefinition->getFactoryClass());
-        $this->assertEquals('factory', $clientDefinition->getFactoryMethod());
+        if (method_exists($clientDefinition, 'getFactory')) {
+            $this->assertEquals(array('Aws\S3\S3Client', 'factory'), $clientDefinition->getFactory());
+        } else {
+            $this->assertEquals('Aws\S3\S3Client', $clientDefinition->getFactoryClass());
+            $this->assertEquals('factory', $clientDefinition->getFactoryMethod());
+        }
```

Comment:
Test must not contains conditional logic. Please add two tests one for old symfony and one for sf2.7